### DISCUSSION TITLE
GenBank Submit Tool

### DISCUSSION THREAD

**Discussion by ccicero** [ArctosDB/arctos#5403] [Announcements]
@dustymc @campmlc @acdoll I want to try using the GenBank submit tool for ~1000 Steller's Jay sequences. Mariel or Andy, I see you have some batches there as test. Has anyone actually successfully submitted to GenBank using this? I am unclear how this works, and we don't seem to have documentation (which I can write if I can figure it out.

Dusty, initial questions/comments:

1) The People fields - you choose an Arctos agent, then need to manually add first_name, middle_initial, last_name. Is that necessary and can't it be done magically by choosing the Arctos agent? What are those separate fields used for?

2) Agent Role - Are 'sequence author' and 'reference author' defined somewhere? And how are these fields used? 

I assume that 'sequence author' is the person generated the sequences, but I'm not sure - and in this case, lots of different students helped with that but are not co-authors. I assume that 'reference author' is a co-author on the publication. If  you choose a role of 'sequence author' then that person also can't be a 'reference author' - ? 

3) Under Sequences, I get this ERROR: syntax error at or near "jion" Position: 223 - I have no idea what that means, but have not tried doing anything yet with the actual sequences.

4) Where do I go from here? I've added the Batch Name, Contact Agent Details, ref_title, and People. Now what do I do???

batch_id=245095978

**Comment by campmlc** [comment]
Our records were just to test, not actually submitted, and we do need a
real test case and documentation.

**Comment by ccicero** [comment]
That's my plan, but I have no idea how this is supposed to work. I have real data and am happy to document, just need some guidance from @dustymc

**Comment by dustymc** [comment]
>People

GenBank wants strange formats - you can save that (as https://arctos.database.museum/info/ctDocumentation.cfm?table=ctaddress_type#formatted_json) so it should be a one-time thing.

>'sequence author' and 'reference author' 

That's just what GB accepts - if they have docs I don't know where.

>ERROR

Patched - so you can add sequences.

I think I'll have to reinstall some software before you can package.

**Comment by ccicero** [comment]
@dusty - I have > 1000 sequences. Do I need to enter them one by one? What would it take to be able to upload a batch of sequences in Fasta file format? This is from the GenBank tool:

![image](https://user-images.githubusercontent.com/967275/132104339-79b1dbfe-64d5-448f-9886-139b33d7c4fe.png)

**Comment by dustymc** [comment]
Send me your data.

**Comment by acdoll** [comment]
> Andy, I see you have some batches there as test

Yes, I was just trying to see how this works. There isn't anything to load there, but I don't see a way to delete. @dustymc can you wipe mine out of there.

**Comment by ccicero** [comment]
@dustymc OK, will do. I am double-checking and need to get the file in fasta format. Are you going to use that as a test for uploading a fasta file, or ??? We could first do a test with just a few sequences manually input, and then try the fasta upload. Or what are you thinking/planning? Thanks.

**Comment by dustymc** [comment]
I"m not thinking anything, I'm just trying to understand. I know very little about how any of this works, or why you might want to use this thing over direct submissions, or - well, anything! Data sometimes help with that.

And yes, if you have individual files I just need a few.

**Comment by ccicero** [comment]
I'm not sure either what the advantage is vs direct submission, but I'll send you the full file and then we can discuss.

I suppose it would be useful if it automatically made the GenBank links, but often the GenBank file will include Arctos and non-Arctos specimens so we'd have to do it in both places anyway - ? I'd like to get input from @campmlc. Maybe we can discuss briefly tomorrow at the AWG meeting?

**Comment by ccicero** [comment]
@dustymc I created loan 9126 (transaction_i=21127705) with 1055 sequences associated with MVZ specimens. I will send you the fasta file separately. I double-checked and the number of sequences in the fasta file should match the specimens in the loan. Sequences are labeled something like '>MX_Chiapas_C1056_MVZ188144' so you can extract the MVZ numbers from the sequence label. 

Feel free to work some magic and let me know if you think this has possibilities that would be useful to Arctos.

If we are going to do this, I have some initial suggestions on the form based on going through the first few Bankit steps in GenBank:

1) Under Contact Agent Details, change 'Organization' to 'Institution' to match the GenBank submission form.

2) Under People: Change heading to 'Sequence Authors' and remove 'reference author' as an option here. All authors should be listed as Sequence Authors.

3) After People, create a new section titled 'Reference' and here:
a) change 'ref_title' to 'Reference Title'
b) add 'Publication Status' with three options: 'unpublished' 'in press' 'published'
![image](https://user-images.githubusercontent.com/967275/132928465-dea93fd6-4a13-45a9-a2a5-287bfb8115d2.png)
c) add 'Reference Authors' here with an option for 'Same as Sequence Authors' or 'Specify New Authors'
![image](https://user-images.githubusercontent.com/967275/132928487-abebbda7-7b28-4c92-92a6-c168e64f2119.png)

Thanks!

**Comment by dustymc** [comment]
@ccicero I'll try to look at this during tomorrow's test outage.

I think I can get to data from this, but I don't think that's something I can push to a UI. I don't know how fasta works, but if something like

```
>MX_Chiapas_C1056_MVZ188144
>MVZ:Bird:188144
>{part_id}
AGT.....
```

or

```
> MX_Chiapas_C1056_MVZ188144|MVZ:Bird:188144|{part_id}

AGT.....
```

or some form of explicit/predictable "headers" is possible/practical then I think I could probably accept that as a file. (I suppose that would need to be part of loan instructions or something???)

One obvious problem is that there's no good way to identify the parts in these data - "some subsample of MVZ104097" could probably be stuffed in some remark somewhere, and would probably get a knowledgeable person where they need to be, but "MVZ104097" is not and can't be used for the actual part identifier. https://github.com/ArctosDB/arctos/issues/3630 provides a solution to that, but @campmlc 's latest comment precludes implementation so I sort of think that idea is just dead. 

So, the packager can...

1. Make sure GenBank gets the record identifiers in the format they can use, but see above - if we're adding that (or some unambiguous proxy to it) to the fasta then that may not be terribly beneficial.
2. Maybe do something interesting by linking into loan data, but lacking persistent partIDs or even barcodes (on the samples) I'm not sure how much value that can have.
3. Make a package for GB submission, but that process is untested and I have no idea how it would compare to the alternative so I can't guess at what value this might have.

Is there any value in that, or is there some other use case where there is value - should we keep this tool alive?

**Comment by ccicero** [comment]
@dustymc Fasta format requires one header line for each sequence, so your first example won't work but the second would be ok. However, there is no predictability to what people put there.

I also think that there is no point in us re-inventing what GenBank does, and the types of data uploaded to GenBank are going to change with new (and current) genomic methods. So I'm not sure that it's worth pursuing a way of batch loading data to GenBank from Arctos. That said, I think we should think about the #2 option. What about something like this as a use case:

- Create a loan that has genetic samples
- Click a button for "Generate NCBI bulkloader"
- Use that to autopopulate an 'other ID' bulkload template with the guid, new_other_id_references
- Generate CSV from that to send with the loan invoice - they only need to then add the new_other_id_type and new_other_id_number. Include something that provides instructions on acceptable new_other_id_type (GenBank, NCBI Biosample).
- Save the bulkloader linked to the loan.
- Have a way of tracking if/when the bulkloader is returned and uploaded

???

**Comment by dustymc** [comment]
>t requires one header line for each sequence

How do you transmit specimen_voucher to GenBank? You've registered MVZ:Bird (https://handbook.arctosdb.org/documentation/genbank.html), they know what to do with that, but I don't think they can dig it out of the sequence name either.

> CSV .. to send with the loan invoice 

That needs an issue so it can have focused discussion, but the technical bits are likely trivial (and it sounds like a pretty great idea to me).

>Have a way of tracking 

https://arctos.database.museum/info/ctDocumentation.cfm?table=ctloan_status#returned
https://arctos.database.museum/info/ctDocumentation.cfm?table=ctloan_status#closed

**Comment by ccicero** [comment]
How do you transmit specimen_voucher to GenBank? You've registered MVZ:Bird (https://handbook.arctosdb.org/documentation/genbank.html), they know what to do with that, but I don't think they can dig it out of the sequence name either.

**--> Correct. It's put in a separate field as part of the upload process. I'm thinking maybe I should upload these via the normal process but take screenshots/documentation along the way so you can see the process. Would that help?**

Closed is currently "All items and products such as reprints and measurements are returned and accounted for." But it is often years before we get a publication or GenBank links or any other loan conditions. One thing that would help is to track whether the signed invoice was returned other than in remarks - adding a field for 'Signed invoice returned' [enter date] would indicate that the transaction was successful (i.e., shipment not lost) and allow us to search for all loans lacking a signed invoice but still keep the loan open if we're waiting on 'products' from the research. 

Another issue is that when we do get pubs or GenBank numbers, they don't have the loan number and it's not always easy to track what loan(s) they came from. Maybe we should add loan number to the ID bulkloader CSV generated from a loan? In fact, we could also use this tool to generate a citation bulkloader to help with the issue of people not citing our specimens. If we generate both a citation bulkloader and ID (NCBI) bulkloader directly from the loan and send them to the recipient along with the invoice, maybe it will push people to properly cite our specimens?!

**Comment by campmlc** [comment]
AND link the GenBank accession and citation to the loan and to the parts in
the loan via a part identifier.

**Comment by dustymc** [comment]
>Would that help?

YES!!

>years before we get a publication or GenBank

Yes - and Arctos sends you reminders!

>'Signed invoice returned' [enter date] 

I think that's just another loan status, but maybe we need shipment status or something too. In any case I'd not want to end up with a bunch of ways to mark progress - needs discussion (and I think there are a bunch of orphaned issues open).

>properly cite 

I'm up for whatever, can certainly generate any paperwork the data will support, but I'm also mostly convinced that these things come back to your loan policies and how you enforce them.

>add loan number to the ID bulkloader CSV generated from a loan?

https://github.com/ArctosDB/arctos/issues/1257

The workaround is to demand users cite parts, which you can do now via https://arctos.database.museum/info/ctDocumentation.cfm?table=ctspecpart_attribute_type#part_identifier or maybe eventually a little more persistently via https://github.com/ArctosDB/arctos/issues/3630

**Comment by ccicero** [comment]
OK, I'll upload via GenBank and document as best as possible.

**Comment by mkoo** [comment]
I wonder if this is worth talking with John Deck since Geome-db has a FAsta module and links to NCBI. It also has the potential to link to other museums (via DwC triples) so if you have a mix of Arctos and non-Arctos collections maybe you could do it in one go (the upload template is an excel file). I admit I am not familiar with the entire process for this linkage since I havent had to do it myself.

**Comment by campmlc** [comment]
That would be useful. The current Arctos model has not had enough testing.

**Comment by ccicero** [comment]
I tried testing but it was confusing. I plan to submit ~1000 sequences to GenBank (hopefully this month!) and told Dusty that I would capture the process with screenshots so that he as a better understanding.

**Comment by dustymc** [comment]
I'm up for anything, but from my perspective the real power of this is that it allows submitting something better than record-level data, and doesn't allow the submission of things that won't resolve - all of which requires a connection to Arctos. (That's currently barcode, https://github.com/ArctosDB/arctos/issues/3630 would be better but there's no buy-in and I think that idea is probably dead.)

>DwC triples

Note that those are **not** the identifier GenBank works with. They should have been, GenBank was onboard, but nothing useful happened so GenBank built their own thing. https://handbook.arctosdb.org/documentation/genbank.html 

There's a list of collections which *have not* registered this with Arctos below.

> havent had to do it myself

Ditto, this seems like it needs some dedicated use case before it gets revived - there's a lot of work in a tool that AFAIK nobody actually uses. 

It also requires some software to be installed, which doesn't work terribly well with the current architecture plan - a dedicated API box would provide a place for that, along with some other stuff.

```
select guid_prefix from collection where genbank_collection is null order by guid_prefix;
  guid_prefix  
---------------
 ACUNHC:Bird
 ACUNHC:Ento
 ACUNHC:Fish
 ACUNHC:Herp
 ACUNHC:Inv
 ACUNHC:Mamm
 ACUNHC:Paleo
 ALMNH:Anc
 ALMNH:Arc
 ALMNH:Bird
 ALMNH:EH
 ALMNH:Ento
 ALMNH:ES
 ALMNH:Geo
 ALMNH:Herp
 ALMNH:Inv
 ALMNH:Mamm
 ALMNH:Paleo
 APSU:Bird
 APSU:Fish
 APSU:Herp
 APSU:Mamm
 Arctos:Entity
 ASNHC:Bird
 ASNHC:Herb
 ASNHC:Herp
 ASNHC:Para
 ASUMZ:Bird
 ASUMZ:Bivalve
 ASUMZ:Ento
 ASUMZ:Fish
 ASUMZ:Herp
 ASUMZ:Invert
 ASUMZ:Mamm
 BYU:Bird
 BYU:Fish
 BYU:Herp
 BYU:Mamm
 BYUObs:Bird
 BYUObs:Herp
 BYUObs:Mamm
 BYU:Teach
 CHAS:Art
 CHAS:Bird
 CHAS:Egg
 CHAS:EH
 CHAS:Ento
 CHAS:ES
 CHAS:Fish
 CHAS:Herb
 CHAS:Herp
 CHAS:Inv
 CHAS:Mamm
 CHAS:Teach
 COA:Ento
 COA:Rept
 DGR:Mamm
 DMNS:Egg
 DMNS:Herp
 DMNS:Inv
 HMCM:Arthro
 HMCM:Bird
 HMCM:Geo
 HMCM:Herb
 HMCM:Herp
 HMCM:Mamm
 JSNM:Egg
 JSNM:Herb
 JSNM:Paleo
 KNWR:Env
 KNWR:Inv
 KNWRObs:Bird
 KNWRObs:Fish
 KNWRObs:Herb
 KNWRObs:Mamm
 LINGU:Herp
 MLZ:Egg
 MLZ:Fish
 MLZ:Herb
 MLZObs:Fish
 MSB:Host
 MSB:Inv
 MSBObs:Herp
 MSBObs:Mamm
 MVZ:ARCH
 MVZ:IMG
 MVZObs:Bird
 MVZObs:Fish
 MVZObs:Herp
 MVZObs:Mamm
 NCWRC:Bird
 NCWRC:Bivalve
 NCWRC:Crust
 NCWRC:Ento
 NCWRC:Fish
 NCWRC:Gastro
 NCWRC:Herp
 NCWRC:Mamm
 NHSM:Arc
 NHSM:Bird
 NHSM:Ento
 NHSM:Geol
 NHSM:Herb
 NHSM:Herp
 NHSM:Inv
 NHSM:Mamm
 NHSM:Paleo
 NMMNH:Bird
 NMMNH:Ento
 NMMNH:Fish
 NMMNH:Geol
 NMMNH:Herb
 NMMNH:Herp
 NMMNH:Inv
 NMMNH:Paleo
 NMU:Bird
 NMU:Ento
 NMU:Fish
 NMU:Herb
 NMU:Herp
 NMU:Mamm
 NMU:Para
 OWU:Amph
 OWU:Bird
 OWU:EH
 OWU:Ento
 OWU:ES
 OWU:Fish
 OWU:Geol
 OWU:Herb
 OWU:Inv
 OWU:Mamm
 OWU:Para
 OWU:Rept
 PSU:Mamm
 STAR:Alg
 STAR:Herb
 TMCC:Bird
 TMCC:Ento
 TMCC:Fish
 TMCC:Herb
 TMCC:Herp
 TMCC:Inv
 TMCC:Mamm
 TMCC:Teach
 UAF:Zooarch
 UAM:Alg
 UAM:Arc
 UAM:Art
 UAMb:Herb
 UAM:EH
 UAM:Env
 UAMObs:Bird
 UAMObs:EH
 UAMObs:Ento
 UAMObs:Fish
 UAMObs:Mamm
 UA:Paleo
 UCM:Egg
 UCM:Obs
 UCONN:Bird
 UCONN:Egg
 UCSC:Bird
 UCSC:Fish
 UC-SCFS:Bird
 UC-SCFS:Mamm
 UCSC:Herp
 UCSC:Mamm
 UMNH:Ento
 UMNH:Mala
 UMNHObs:Mamm
 UMNH:Teach
 UMZM:Bird
 UMZM:Mamm
 UNCG:Mamm
 UNCG:Para
 UNM:MET
 UNM:MIN
 UNM:Paleo
 UNR:Bird
 UNR:Egg
 UNR:Fish
 UTEP:Arc
 UTEP:HerpOS
 UTEPObs:Ento
 UTEPObs:Herp
 UTEP:Teach
 UWYMV:Egg
 UWZM:Bird
 UWZM:EH
 UWZM:Fish
 UWZM:Herp
 UWZM:Inv
 UWZM:Mamm
 UWZM:Paleo
 UWZM:Teach
(196 rows)
```

**Comment by dustymc** [comment]
And here are collections that do not have a registered GenBank identifier and do have GenBank OtherIDs.

```
select guid_prefix from 
collection
inner join cataloged_item on collection.collection_id=cataloged_item.collection_id 
inner join coll_obj_other_id_num on cataloged_item.collection_object_id=coll_obj_other_id_num.collection_object_id and other_id_type='GenBank'
 where genbank_collection is null group by guid_prefix order by guid_prefix;


 BYU:Herp
 KNWR:Inv
 MSB:Host
 UAMObs:Ento

```

**Comment by Jegelewicz** [comment]
> a registered GenBank identifier

What is this and how does a collection get one? I think @ccicero was looking into this and discovered it was no longer a thing? We removed the "register with Genbank" step from data migration because of that discussion which I cannot find.

**Comment by dustymc** [comment]
>What is this and how does a collection get one? I

https://github.com/ArctosDB/arctos/issues/2413

>no longer a thing

That's a question for GenBank, but I can't imagine how a few things could work without it.

**Comment by campmlc** [comment]
Do we have documentation on how to get a collection registered at GenBank?

**Comment by Jegelewicz** [comment]
No we don't and as many times as I've asked I have never figured out exactly what that means or how to get it done....

**Comment by dustymc** [comment]
I don't know what the process is either, but I'm happy to help figure it out if there's a collection in need of being registered.

**Comment by campmlc** [comment]
Looks like a lot of collections, from your list.

**Comment by jrdemboski** [comment]
you might contact the Curator of Biosciences at NCBI, that's who's recently handled any collections registration for me or updating institutional info, etc.

Shobha Sharma, PhD
Staff Scientist
GenBank Taxonomy
NCBI/NLM/NIH
***@***.***
[https://training.knowbe4.com/pab_uploads_prefix/pab_icon/6661FF0593A413FEE166DD9DE2FFE18C/icon_64x64.png]






To:

  *
 John Demboski

Cc:

  *
 Sharma, Shobha (NIH/NLM/NCBI) [E] ***@***.***>




________________________________

**Comment by ewommack** [comment]
What about updating collection information on Genbank? I just checked the UWYMV, and while our main collections are there, our contact information is out of date.

**Comment by dustymc** [comment]
I don't know of any way I can update either - you can try emailing linkout at ncbi.nlm.nih.gov or John's contact above.

**Comment by jrdemboski** [comment]
Go directly to Shobha for institutional updates and registering collections:

Shobha Sharma, PhD
Staff Scientist
GenBank Taxonomy
NCBI/NLM/NIH
***@***.******@***.***>



From: dustymc ***@***.***>
Date: Thursday, February 10, 2022 at 2:08 PM
To: ArctosDB/arctos ***@***.***>
Cc: John Demboski ***@***.***>, Comment ***@***.***>
Subject: Re: [ArctosDB/arctos] GenBank Submit Tool (#3903)

I don't know of any way I can update either - you can try emailing linkout at ncbi.nlm.nih.gov or John's contact above.

—
Reply to this email directly, view it on GitHub<https://github.com/ArctosDB/arctos/issues/3903#issuecomment-1035519291>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ABLZA2UBQSXP3OARXF6BRCLU2QSNRANCNFSM5DMVD26A>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.

**Comment by ccicero** [comment]
This issue has split into two: GenBank submit tool, and registering with GenBank.

Re: the submit tool, I recently went through a GenBank submission for >1000 sequences and I don't really see how we can develop a tool for that. Moreover, what are we trying to accomplish? It's a complicated process and lots of different options depending on the type of data. I tried to take screenshots (see below). I think the tools we have for looking for GenBank numbers that might match Arctos records are great, but I can't see what we want to try and do beyond that. My two cents, but happy to discuss further.

[BankIt_screenshots.pdf](https://github.com/ArctosDB/arctos/files/8319628/BankIt_screenshots.pdf)

**Comment by Jegelewicz** [comment]
> Go directly to Shobha for institutional updates and registering collections:
> 
> Shobha Sharma, PhD
> Staff Scientist
> GenBank Taxonomy
> NCBI/NLM/NIH

@jrdemboski can you send me her email via email so that I can have it to give to new collections that want to do this?

**Comment by ccicero** [comment]
I wonder if we can somehow work this into Manage Collection. Have options for 'Register with Genbank' and whatever is required to do that, and somehow get this information to GenBank (Shobha for now, but something more generic so that it's not dependent on a single person)? It would be helpful to know what they need for this.

**Comment by Jegelewicz** [comment]
> It would be helpful to know what they need for this.

Yes it would.

**Comment by campmlc** [comment]
Agree. I foubd Shoba's email
sramanadhan@hsph.harvard.edu.
